TestCase/Managers/GameManager.cs

```diff
@@ -0,0 +1,100 @@
+﻿namespace TestCase.Managers
+{
+    public class GameManager
+    {
+        private readonly BoardManager _boardManager;
+        private readonly InputManager _inputManager;
+
+        private bool isPlayerXTurn = true;
+        private bool isPlayerOTurn = false;
+
+        private int turnCounter = 0;
+
+        public GameManager()
+        {
+            this._boardManager = new BoardManager();
+            this._inputManager = new InputManager();
+        }
+
+        public void InputFromUser()
+        {
+            if (isPlayerXTurn)
+            {
+                Console.WriteLine("\nit is currently Player X's turn");
+            }
+            else if (isPlayerOTurn)
+            {
+                Console.WriteLine("\nit is currently Player O's turn");
+            }
+
+            int row = _inputManager.GetInput() - 1;
+
+            int col = _inputManager.GetInput() - 1;
+
+            PlayerTurn(row, col);
+        }
+
+        public void PlayerTurn(int row, int col)
+        {
+           if (_boardManager.PlaceToken(row, col,isPlayerXTurn ? GlobalConsts.X_PLAYER_TOKEN : GlobalConsts.O_PLAYER_TOKEN)) 
+            {
+                isPlayerXTurn = !isPlayerXTurn;
+                isPlayerOTurn = !isPlayerOTurn;
+            }
+            else 
+            {
+                Console.WriteLine("\nSpace taken try again");
+                InputFromUser();
+            }
+        }
+
+        public void ResetGame()
+        {
+            Console.WriteLine("\n++++++++++++++++++++++++++++++NEW GAME++++++++++++++++++++++++++++++++++++++");
+            turnCounter = 0;
+
+            isPlayerXTurn = true;
+            isPlayerOTurn = false;
+
+            _boardManager.ResetBoard();
+
+            StartGame();
+
```

Comment:
It looks like we set the values of these class variables, twice (once when declaring and again here):
- `turnCounter`
- `isPlayerXTurn`
- `isPlayerOTurn`

To avoid repeated code, could we have a private method, something like `ConfigureNewGame()` or similar. This would set the above values and call `_boardManager.ResetBoard()`. Can call this in the constructor and here and remove the variable initlising at the top.

Benefit of this is if we want to change the settings it is in one place, so 1 change, not 2.

Reply:
Something similar would need to be done in `BoardManager.cs`, where you separate out the variable init from declaration. So we can call it here and it wont rewrite on app start. Could call it `InitialiseBoard()`, makes sense for both game start and restart. Obv could have a better name :)